numba/np/arraymath.py

```diff
@@ -4722,3 +4722,36 @@ def impl(a, b):
         return _cross2d_operation(a_, b_)
 
     return impl
+
+
+@overload(np.trim_zeros)
+def np_trim_zeros(a, trim='fb'):
+    if not isinstance(a, types.Array):
+        raise NumbaTypeError('The first argument must be an array-like')
+
+    if hasattr(a, 'ndim') and a.ndim > 1:
+        raise NumbaTypeError('array must be 1D')
```

Comment:
`a` should always have an `ndim` attr, if it's always an array:

```suggestion
    if a.ndim > 1:
        raise NumbaTypeError('array must be 1D')
```